packages/components/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,105 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { cloneElement, forwardRef, type MouseEvent, type ReactElement, type ReactNode } from 'react';
+
+import { buttonIconSizeMap, buttonStyles } from './styles/buttonStyles';
+import { iconStyles } from './styles/iconStyles';
+import { textStyles } from './styles/textStyles';
+
+export type ButtonRounding = 'medium' | 'full';
+
+export type ButtonStyle = 'default' | 'positive' | 'negative' | 'danger' | 'loud';
+
+export type UpdatedButtonStyle = 'default' | 'positive' | 'negative' | 'danger' | 'loud';
+
+export type ButtonSize = 'small' | 'medium' | 'large';
+
+export type ButtonType = 'button' | 'submit' | 'reset';
+
+export type ButtonEmphasis = 'default' | 'weak' | 'strong';
+
+export type ButtonProps = {
+    type?: ButtonType;
+    title?: string;
+    style?: ButtonStyle;
+    emphasis?: ButtonEmphasis;
+    hideLabel?: boolean;
+    size?: ButtonSize;
+    rounding?: ButtonRounding;
+    disabled?: boolean;
+    icon?: ReactElement;
+    children?: ReactNode;
+    onClick?: (event?: MouseEvent<HTMLButtonElement>) => void;
+    hugWidth?: boolean;
+    'aria-label'?: string;
+    'aria-describedby'?: string;
+    formId?: string;
+    'data-test-id'?: string;
+};
+
+export const ButtonComponent = (
+    {
+        type = 'button',
+        style = 'default',
+        size = 'medium',
+        rounding = 'medium',
+        emphasis = 'strong',
+        title = '',
+        hideLabel = false,
+        disabled = false,
+        icon,
+        children,
+        onClick,
+        hugWidth = true,
+        formId,
+        'aria-label': ariaLabel,
+        'aria-describedby': ariaDescribedBy,
+        'data-test-id': dataTestId = 'button',
+    }: ButtonProps,
+    ref: React.ForwardedRef<HTMLButtonElement | null>,
+) => {
+    return (
+        <button
+            data-test-id={dataTestId}
+            aria-label={ariaLabel}
+            aria-disabled={disabled}
+            aria-describedby={ariaDescribedBy}
+            ref={ref}
+            form={formId}
+            title={title}
+            type={type}
+            onClick={onClick}
+            disabled={disabled}
+            className={buttonStyles({
+                disabled,
+                rounding,
+                size,
+                hugWidth,
+                emphasis,
+                style,
+                iconOnly: (icon && !children) || hideLabel,
+            })}
+        >
+            {icon && (
```

Comment:
Icon should not be known by the Button component, please remove it and maybe we can add a size `icon` or `square` which make the button the same width and height.

Reply:
how do you mean? you don't want to pass in the icon as a prop to the button?

Reply:
Exactly, IMO should be
```tsx
<Button size="icon"> // or square
  <IconABC size={20} />
</Button>
```

Reply:
So we can avoid the `cloneElement`

Reply:
let's have a chat about this tomorrow